woocommerce/payment-gateway/admin/class-sv-wc-payment-gateway-admin-order.php

```diff
@@ -0,0 +1,439 @@
+
+			$order = wc_get_order( SV_WC_Helper::get_request( 'post' ) ); // TODO: best way to get the order?
+
+			if ( ! $order ) {
+				return;
+			}
+
+			wp_localize_script( 'sv-wc-payment-gateway-admin-order', 'sv_wc_payment_gateway_admin_order', array(
+				'ajax_url'       => admin_url( 'admin-ajax.php' ),
+				'gateway_id'     => SV_WC_Order_Compatibility::get_prop( $order, 'payment_method' ),
+				'order_id'       => SV_WC_Order_Compatibility::get_prop( $order, 'id' ),
+				'capture_ays'    => __( 'Are you sure you wish to process this capture? The action cannot be undone.', 'woocommerce-plugin-framework' ),
+				'capture_action' => 'wc_' . $this->get_plugin()->get_id() . '_capture_charge',
+				'capture_nonce'  => wp_create_nonce( 'wc_' . $this->get_plugin()->get_id() . '_capture_charge' ),
+				'capture_error'  => __( 'Something went wrong, and the capture could no be completed. Please try again.', 'woocommerce-plugin-framework' ),
+			) );
+
+			wp_enqueue_style( 'sv-wc-payment-gateway-admin-order', $this->get_plugin()->get_payment_gateway_framework_assets_url() . '/css/admin/sv-wc-payment-gateway-admin-order.min.css', SV_WC_Plugin::VERSION );
+		}
+	}
+
+
+	/** Capture Charge Feature ******************************************************/
+
+
+	/**
+	 * Adds 'Capture charge' to the Orders screen bulk action select.
+	 *
+	 * @since 5.0.0-dev
+	 */
+	public function maybe_add_capture_charge_bulk_order_action() {
+		global $post_type, $post_status;
+
+		if ( $post_type === 'shop_order' && $post_status !== 'trash' ) {
+
+			$can_capture_charge = false;
+
+			// ensure at least one gateway supports capturing charge
+			foreach ( $this->get_plugin()->get_gateways() as $gateway ) {
+
+				// ensure that it supports captures
+				if ( $gateway->supports_credit_card_capture() ) {
+
+					$can_capture_charge = true;
+					break;
+				}
+			}
+
+			if ( $can_capture_charge ) {
+
+				?>
+					<script type="text/javascript">
+						jQuery( document ).ready( function ( $ ) {
+							if ( 0 == $( 'select[name^=action] option[value=wc_capture_charge]' ).size() ) {
+								$( 'select[name^=action]' ).append(
+									$( '<option>' ).val( '<?php echo esc_js( 'wc_capture_charge' ); ?>' ).text( '<?php _e( 'Capture Charge', 'woocommerce-plugin-framework' ); ?>' )
+								);
+							}
+						});
+					</script>
+				<?php
+			}
+		}
+	}
+
+
+	/**
+	 * Process the 'Capture Charge' custom bulk action on the Orders screen
+	 * bulk action select
+	 *
+	 * @since 2.1.0
+	 */
+	public function process_capture_charge_bulk_order_action() {
+		global $typenow;
+
+		if ( 'shop_order' == $typenow ) {
+
+			// get the action
+			$wp_list_table = _get_list_table( 'WP_Posts_List_Table' );
+			$action        = $wp_list_table->current_action();
+
+			// bail if not processing a capture
+			if ( 'wc_capture_charge' !== $action ) {
+				return;
+			}
+
+			// security check
+			check_admin_referer( 'bulk-posts' );
```

Comment:
should we also perform a permissions check like `current_user_can( 'manage_woocommerce' )` here as well? If so, probably applies to our other bulk actions as well.